src/app/modules/did-book/components/select-did/select-did.component.ts

```diff
@@ -1,54 +1,104 @@
-import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
 import { FormControl, Validators } from '@angular/forms';
-import { HexValidators } from '../../../../utils/validators/is-hex/is-hex.validator';
+import { HexValidators } from '@utils';
 import { DidBookService } from '../../services/did-book.service';
-import { combineLatest } from 'rxjs';
+import { combineLatest, Observable } from 'rxjs';
 import { map, startWith } from 'rxjs/operators';
 import { MatDialog } from '@angular/material/dialog';
 import { AddSingleRecordComponent } from '../add-single-record/add-single-record.component';
+import { ListValidator } from '@utils';
+import { DidBookRecord } from '../models/did-book-record';
 
 @Component({
   selector: 'app-select-did',
   templateUrl: './select-did.component.html',
   styleUrls: ['./select-did.component.scss'],
 })
 export class SelectDidComponent implements OnInit {
+  @Input() label: string;
+  @Input() placeholder: string;
+  @Input() isRequired: boolean = false;
+  @Input() showAddButton: boolean;
+  @Input() set didToRemove(list: string[]) {
+    this._didToRemove = list;
+    this.setValidators();
+    this.setDIDBookList();
+  }
+  get didToRemove() {
+    return this._didToRemove;
+  }
   @Output() didChange = new EventEmitter<{ did: string; valid: boolean }>();
+  @Output() add = new EventEmitter<string>();
 
-  didBook$;
-  newOwnerDID = new FormControl('', [
-    Validators.required,
-    HexValidators.isDidValid(),
-  ]);
-  isNotKnownDid: boolean;
+  didBook$: Observable<DidBookRecord[]>;
+  newDID = new FormControl('', [HexValidators.isDidValid()]);
+  isNotKnownDid: boolean = false;
+  private _didToRemove: string[] = [];
 
   constructor(private didBookServ: DidBookService, private dialog: MatDialog) {}
 
   ngOnInit(): void {
+    this.setValidators();
+    this.setDIDBookList();
+
+    this.newDID.valueChanges.subscribe((did) => {
+      this.didChange.emit({ did, valid: this.newDID.valid });
+    });
+  }
+
+  setValidators(): void {
+    if (!this.isRequired) {
+      this.newDID.setValidators([...this.defaultValidators()]);
+      return;
+    }
+    this.newDID.setValidators([
+      Validators.required,
+      ...this.defaultValidators(),
+    ]);
+  }
+
+  approveHandler(): void {
+    this.dialog.open(AddSingleRecordComponent, {
+      width: '550px',
```

Comment:
Is this kind of width param hardcoded elsewhere in the app? I would think that this might make the app less usable on mobile 🤔 ?

Reply:
No, it looks fine on small viewports. I've checked it. When viewport is smaller than 550px dialog is adjusting. However when viewport is bigger, then it is not wider than 550px.